### PR TITLE
Fix String Comparison to use POSIX standard #2481

### PR DESCRIPTION
**Pull request by peterskipper**
# Pull Request Template

## Description
Making the conda install script compatible with more flavors of shell. `==` is not valid in some shells for string comparison, and `=` conforms to POSIX!
Fix #(2481)

This PR fixes issue #2481 . It uses a single `=` to compare strings which is valid across a wider range of shells.

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterskipper** [comment]
Hmm not sure why the tests are failing. I definitely didn't change any of that code!

**Comment by peastman** [comment]
Don't worry about that test.  It's currently failing on the main branch.  The fix is in #2486.

This looks fine to me.

**Review by alat-rights — APPROVED** [review]

**Comment by peterskipper** [comment]
This is great! Excited to make my first (tiny) contribution. It appears as though the change has been approved, but I believe I don't have permission to merge it in. Are there any additional steps?

**Comment by alat-rights** [comment]
Oh gosh I'm sorry I'm not actually a maintainer. Maybe if you could wait for one of the maintainers to approve?

**Review by rbharath — APPROVED** [review]
LGTM!

**Pull request merged**
